plugins/CuraEngineBackend/CuraEngineBackend.py

```diff
@@ -301,41 +307,58 @@ def _onStartSliceCompleted(self, job):
 
         Logger.log("d", "Sending slice message took %s seconds", time() - self._slice_start_time )
 
+    ##  Determine enable or disable auto slicing. Return True for enable timer and False otherwise.
+    #   It disables when
+    #   - preference auto slice is off
+    #   - decorator isBlockSlicing is found (used in g-code reader)
+    def determineAutoSlicing(self):
+        enable_timer = True
+
+        if not Preferences.getInstance().getValue("general/auto_slice"):
+            enable_timer = False
+        for node in DepthFirstIterator(self._scene.getRoot()):
+            if node.callDecoration("isBlockSlicing"):
+                enable_timer = False
+                self.backendStateChange.emit(BackendState.Disabled)
+            gcode_list = node.callDecoration("getGCodeList")
+            if gcode_list is not None:
+                self._scene.gcode_list = gcode_list
+
+        if self._use_timer == enable_timer:
+            return
```

Comment:
Return either true or false. None correct in most cases, but not if someone explicitly decides to check for determineAutoSlicing() == False
If this is intentional, please update the function documentation accordingly.